public/app/features/alerting/unified/components/rule-editor/ExpressionEditor.tsx

```diff
@@ -41,20 +54,52 @@ export const ExpressionEditor: FC<ExpressionEditorProps> = ({ value, onChange, d
     return <div>Could not load query editor due to: {errorMessage}</div>;
   }
 
+  const previewLoaded = alertPreview?.data.state === LoadingState.Done;
+  // The preview API returns arrays with empty elements when there are no firing alerts
+  const previewHasAlerts = alertPreview?.data.series.every((frame) =>
+    frame.fields.every((field) => field.values.length > 0)
+  );
+
   const QueryEditor = dataSource?.components?.QueryEditor;
 
+  const previewSeries = alertPreview?.data.series ?? [];
+
   return (
-    <QueryEditor
-      query={query}
-      queries={[query]}
-      app={CoreApp.CloudAlerting}
-      onChange={onChangeQuery}
-      onRunQuery={noop}
-      datasource={dataSource}
-    />
+    <>
+      <QueryEditor
+        query={dataQuery}
+        queries={[dataQuery]}
+        app={CoreApp.CloudAlerting}
+        onChange={onChangeQuery}
+        onRunQuery={noop}
+        datasource={dataSource}
+      />
+
+      <div className={styles.preview}>
+        <Button type="button" onClick={onRunQueriesClick} disabled={alertPreview?.data.state === LoadingState.Loading}>
+          Preview alerts
+        </Button>
+        {previewLoaded && !previewHasAlerts && (
+          <Alert title="Alerts preview" severity="info" className={styles.previewAlert}>
+            There are no firing alerts for your query.
+          </Alert>
+        )}
+        {previewHasAlerts && <CloudAlertPreview previewSeries={previewSeries} />}
+      </div>
+    </>
   );
 };
 
+const getStyles = (theme: GrafanaTheme2) => ({
+  preview: css`
+    padding: ${theme.spacing(2, 0)};
+    max-width: ${theme.breakpoints.values.md}px;
```

Comment:
I think this could go a bit larger, if you have lots of labels you could definitely use the space :D